internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -1137,6 +1137,14 @@ func upgradeSettingsSchema() *pluginsdk.Schema {
 						Type:     pluginsdk.TypeString,
 						Required: true,
 					},
+					"drain_timeout_in_minutes": {
+						Type:     pluginsdk.TypeInt,
+						Optional: true,
+					},
+					"node_soak_duration_in_minutes": {
+						Type:     pluginsdk.TypeInt,
+						Optional: true,
+					},
```

Comment:
Same here, docs indicate this can be set between 0 and 30 minutes, so we should add that as validation here.